tools/gather-sysinfo/gather-sysinfo.go

```diff
@@ -44,9 +47,27 @@ func newSnapshotCommand(knitOpts *cmd.KnitOptions) *cobra.Command {
 	// use this to debug container behaviour
 	snap.Flags().IntVar(&opts.sleepTime, "sleep", 0, "amount of seconds to sleep once done, before exit")
 
+	collectMachineinfo()
+
 	return snap
 }
 
+func collectMachineinfo() {
+	err := os.MkdirAll("/etc/machineinfo/", os.ModePerm)
```

Comment:
I recommended this path, but on retrospected it seems less and less nice to temper `/etc`, so I'd just create `machineinfo.json` in `/var/log`. This way we save ourselves a `MkdirAll`.